beancount/utils/misc_utils.py

```diff
@@ -281,15 +282,15 @@ def compute_unique_clean_ids(strings):
     return idmap
 
 
-def escape_string(string):
+def escape_string(string: str) -> str:
     """Escape quotes and backslashes in payee and narration.
 
     Args:
       string: Any string.
     Returns.
       The input string, with offending characters replaced.
     """
-    return string.replace("\\", r"\\").replace('"', r"\"")
+    return json.dumps(string, ensure_ascii=False)[1:-1]
```

Comment:
Why?

Reply:
> we parse `r"\n"` as `"\n"` but we didn't escape it back when printing. even we have a `test_escape_string` for that but it has wrong assert.

Reply:
I don't think there is a problem here. If there is a problem, how does the test work on platforms using `\n` as native newline? Anyhow, why use `json.dumps()`? I start to have the feeling that there is AI crap involved here...

Reply:
Please look at your patch: you change the implementation of `escape_string()` and then change the test to match your implementation. There test is correct for the current implementation. The current code implements the desired behavior. Why is this behavior not the correct one, in your opinion?

Reply:
because I don't want to write double quoting escaping logic again.

And this is not a problem before because we didn't notice here, no one realy use `\\n` inside double quoted string.

`print_entries` doesn't revers paring as expected, try this example and see output from your console (different from before)

```
from beancount import print_entries
from beancount.loader import load_string

row = r"""
2014-06-10 * "Entry with escaped \"symbols\" \ \r \n"
  Assets:Account1       111.00 BEAN
  Assets:Cash          -111.00 BEAN
"""

print_entries(load_string(row)[0])
"""

Reply:
What the function does is explained in the doc string? Did you read it? At a minimum, if you change what the function does you need to change the doc string as well.

Reply:
> because I don't want to write double quoting escaping logic again

What makes you think that the escape sequences accepted by the `json` module are the same accepted by the beancount parser?